### PR TITLE
qa/suite/crimson-rados: Fix rados_python test

### PR DESCRIPTION
**Pull request by Matan-B**
* `test_remove_omap_range2`: The failure was caused by inconsistent behavior of cyanstore in comparison to our other backends. `CyanStore::_omap_rmkeyrange` was updated to not remove the last key in the range.
* `test_aio_read_wait_for_complete_*`: These (3) tests excluded are currently hanging with crimson-osd due to Objecter's (un-futurized) call to monc->get_version(). 
* `rados_python.yaml`: This test only runs for few minutes, timeout was reduced from 3h to 1h to avoid hanging jobs.

`rados:basic` run: https://pulpito.ceph.com/matan-2022-08-17_06:20:56-rados:basic-main-distro-default-smithi/
```
2022-08-11T06:07:14.250 INFO:tasks.workunit.client.0.smithi116.stderr:Ran 73 tests in 231.378s
2022-08-11T06:07:14.250 INFO:tasks.workunit.client.0.smithi116.stderr:
2022-08-11T06:07:14.250 INFO:tasks.workunit.client.0.smithi116.stderr:OK
```

`crimson-rados` run: https://pulpito.ceph.com/matan-2022-08-11_08:04:16-crimson-rados-main-distro-default-smithi/
`smoke:basic` run: https://pulpito.ceph.com/matan-2022-08-17_06:21:40-smoke:basic-main-distro-default-smithi/

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by NitzanMordhai — APPROVED** [review]
LGTM

**Comment by athanatos** [comment]
retest this please

**Review by athanatos — APPROVED** [review]

**Comment by Matan-B** [comment]
jenkins test api

File: qa/suites/crimson-rados/basic/tasks/rados_python.yaml [hunk 1]
```diff
@@ -12,4 +12,4 @@ tasks:
 - workunit:
     clients:
       client.0:
-        - rados/test_python.sh --eval-attr 'not (tier or snap or ec or bench or stats)'
+        - rados/test_python.sh --eval-attr 'not (wait or tier or snap or ec or bench or stats)'
```
Comment (rzarzynski) [inline qa/suites/crimson-rados/basic/tasks/rados_python.yaml:16]: > wait_for_complete tests currently hanging with crimson-osd due to
> Objecter's (un-futurized) call to monc->get_version().

Why we would need to futurize `Objecter` or things it calls? Isn't that a client-side code?

**Review by rzarzynski — APPROVED** [review]
LGTM when it comes to code changes. One question about a commit description.

File: qa/suites/crimson-rados/basic/tasks/rados_python.yaml (hunk 1 above)
Comment (Matan-B) [reply qa/suites/crimson-rados/basic/tasks/rados_python.yaml:16]: `Objecter` uses (e.g in `wait_for_latest_osdmap()`) Classic's MonClient get_version (`monc->get_version("osdmap", CB_Objecter_...`).
Crimson's MonClient get_version() is from `seastar::future<....> Client::get_version()` type. We should support Objecter's usage as well.

Comment (rzarzynski) [reply qa/suites/crimson-rados/basic/tasks/rados_python.yaml:16]: Why? Do we have any potential user for it inside crimson-osd? Things external to crimson (e.g. tests, RBD; RADOS clients in general) should be fine with the classical `MonClient` IIUC.

Comment (Matan-B) [reply qa/suites/crimson-rados/basic/tasks/rados_python.yaml:16]: Updated the commit message after our discussion.

**Review by rzarzynski — APPROVED** [review]

**Comment by Matan-B** [comment]
jenkins test api

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
jenkins test api

**Comment by Matan-B** [comment]
jenkins test api

**Pull request merged**